src/components/drawer/drawer.component.js

```diff
@@ -21,20 +28,78 @@ const Drawer = ({
   animationDuration,
   ...props
 }) => {
-  const isControlled = expanded !== undefined;
+  const drawerSidebarContentRef = useRef();
+  const isControlled = useRef(expanded !== undefined);
   const [isExpandedInternal, setIsExpandedInternal] = useState(defaultExpanded || false);
-  const [isFirstOpen, setFirstOpen] = useState(true);
-  const isExpanded = isControlled ? expanded : isExpandedInternal;
+  const [isOpening, setIsOpening] = useState(false);
+  const [isClosing, setIsClosing] = useState(false);
+  const isExpanded = isControlled.current ? expanded : isExpandedInternal;
+  const timer = useRef();
+
+  useEffect(() => {
+    const message = 'Drawer should not switch from uncontrolled to controlled'
+    + ' (or vice versa). Decide between using a controlled or uncontrolled Drawer element'
+    + ' for the lifetime of the component';
+    invariant(isControlled.current === (expanded !== undefined), message);
+  }, [expanded]);
```

Comment:
i think that you could remove `isExpandedInternal` and just set it in the useEffect

```
  const [isExpanded, setIsExpanded] = useState(false);

  useEffect(() => {
  const newValue = expanded
    const message = 'Drawer should not switch from uncontrolled to controlled'
    + ' (or vice versa). Decide between using a controlled or uncontrolled Drawer element'
    + ' for the lifetime of the component';
    invariant(isControlled.current === (expanded !== undefined), message);
    
    setIsExpanded(expanded);
  }, [expanded]);
```

Reply:
👍

Reply:
I forgot to add defaultExpanded in that comment, I had to edit that